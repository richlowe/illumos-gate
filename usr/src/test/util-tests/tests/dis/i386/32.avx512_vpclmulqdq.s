/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2020 Robert Mustacchi
 */

/*
 * Test AVX512 vpclmulqdq related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vpclmulqdq	$0x23, %ymm0, %ymm1, %ymm2
	vpclmulqdq	$0x42, 0x12345(%eax), %ymm3, %ymm4
	vpclmulqdq	$-0x17, 0x678(%ebx, %ecx, 4), %ymm3, %ymm4

	vpclmulqdq	$0x23, %zmm0, %zmm1, %zmm2
	vpclmulqdq	$0x42, 0x12345(%eax), %zmm3, %zmm4
	vpclmulqdq	$-0x17, 0x678(%ebx, %ecx, 4), %zmm3, %zmm4
.size libdis_test, [.-libdis_test]
