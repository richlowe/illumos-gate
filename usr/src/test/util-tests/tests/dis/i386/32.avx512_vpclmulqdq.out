    libdis_test:      c4 e3 75 44 d0 23  vpclmulqdq $0x23,%ymm0,%ymm1,%ymm2
    libdis_test+0x6:  c4 e3 65 44 a0 45  vpclmulqdq $0x42,0x12345(%eax),%ymm3,%ymm4
                      23 01 00 42 
    libdis_test+0x10: c4 e3 65 44 a4 8b  vpclmulqdq $-0x17,0x678(%ebx,%ecx,4),%ymm3,%ymm4	<0xe9>
                      78 06 00 00 e9 
    libdis_test+0x1b: 62 f3 75 48 44 d0  vpclmulqdq $0x23,%zmm0,%zmm1,%zmm2
                      23 
    libdis_test+0x22: 62 f3 65 48 44 a0  vpclmulqdq $0x42,0x12345(%eax),%zmm3,%zmm4
                      45 23 01 00 42 
    libdis_test+0x2d: 62 f3 65 48 44 a4  vpclmulqdq $-0x17,0x678(%ebx,%ecx,4),%zmm3,%zmm4	<0xe9>
                      8b 78 06 00 00 e9 
